lib/routes/builds.js

```diff
@@ -144,24 +144,40 @@ module.exports = function (app) {
     //
     var specs = spec(req.params, req.query);
 
-    //
-    // If we have a version, do search, otherwise fetch from HEAD,
-    // we may want to automatically handle this in `bffs`
-    //
-    return specs.version
-      ? app.bffs.search(specs, fetched)
-      : app.bffs.head(specs, fetched);
+    return fetch(specs);
+
+    function notFound() {
+      error = new Error('Build not found');
+      error.status = 404;
+      return next(error);
+    }
+
+    function fetch(spec) {
+      //
+      // If we have a version, do search, otherwise fetch from HEAD,
+      // we may want to automatically handle this in `bffs`
+      //
+      return spec.version
+        ? app.bffs.search(spec, fetched)
+        : app.bffs.head(spec, fetched);
+    }
 
     function fetched(err, build) {
       if (err) {
         err.status = 500;
         return next(err);
       }
 
-      if (!build) {
-        error = new Error('Build not found');
-        error.status = 404;
-        return next(error);
+      //
+      // Remark: Do a secondary lookup when we have a locale if we get a miss the first
+      // time if we have a 2 part locale
+      //
+      if (!build && !specs.locale) return notFound();
+      else if (!build && specs.locale) {
+        let parts = specs.locale.split('-');
+        if (parts.length !== 2) return notFound();
```

Comment:
@indexzero yea but that wouldn't hit this retry case, that would have been what was requested initially. this would handle fallback from `en-US` to `en` or `es-ES` to `es`